homeassistant/components/rpi_gpio/__init__.py

```diff
@@ -53,8 +53,15 @@ def read_input(port):
     return GPIO.input(port)
 
 
-def edge_detect(port, event_callback, bounce):
+def edge_detect(port, edge, event_callback, bounce):
     """Add detection for RISING and FALLING events."""
     from RPi import GPIO  # pylint: disable=import-error
 
-    GPIO.add_event_detect(port, GPIO.BOTH, callback=event_callback, bouncetime=bounce)
+    if edge == "BOTH":
+        edge = GPIO.BOTH
+    elif edge == "RISING":
+        edge = GPIO.RISING
+    elif edge == "FALLING":
+        edge = GPIO.FALLING
```

Comment:
Use the `DetectEnum(Enum)`. You can also offload it into `rpi_gpio/const.py`

Reply:
please address this comment

Reply:
This module is an abstraction of the RPi GPIO functions, I would not want to pollute it with `DetectEnum`, which belongs to the binary sensor only.